aplib.net-demo/Assets/Scripts/Entities/Enemy/MeleeEnemy.cs

```diff
@@ -2,32 +2,45 @@
 using System.Collections;
 using UnityEngine;
 
-[RequireComponent(typeof(Timer))]
 /// <summary>
 /// Melee enemy that attacks the player when in range.
 /// It inherits from DummyEnemy to add respawn functionality.
 /// </summary>
+[RequireComponent(typeof(Timer))]
+[RequireComponent(typeof(EntitySound))]
 public class MeleeEnemy : DummyEnemy
 {
-    [SerializeField] private float _attackCooldown = 2f;
-    [SerializeField] private float _hitDelay = 1f;
+    [SerializeField]
+    private float _attackCooldown = 2f;
+
+    [SerializeField]
+    private float _hitDelay = 1f;
+
+    [SerializeField]
+    private float _swingWidth = 1.4f;
+
+    [SerializeField]
+    private float _swingLength = 4f;
+
     private readonly float _sizeIncrease = 1.2f;
     private MeleeWeapon _meleeWeapon;
     private Timer _cooldownTimer;
     private bool _isSwinging = false;
+    private EntitySound _entitySound;
```

Comment:
We do not want to store the EntitySound and use it in this class, requiring it to enforce it existing on an enemy is fine, but it should be used within the weapons as they were, so that they can be used more easily and do not have to be used for every enemy type we may introduce

(Right now it's easier, melee weapon = melee sound, ranged weapon = ranged sound, check this out on the main branch!)

this comment applies to all uses of entity sound in this script

Reply:
Yes I agree for the weapons but this is for the OnDeath soundeffects. In hindsight, I disagree with the way death was done in the sound PR. I think it is cleaner for the death soundeffects to be handled in the enemies themselves then for EntitySound to require a HealthComponent and listen to the death event. This way you can still have enemies with sound (from the weapons) but without a death sound .

Reply:
I somewhat agree with your point @JoenvandeVorle; the entities shouldn't always implement a death sound. I don't think it should be done in the enemies themselves either, because in my opinion that violates the modular architecture that Unity supports. Basically, the EntitySound script should make this more flexible (allowing people to link sound clips to events through the inspector, ideally). The EntitySound script is a temporary fix, however, as sound clips will be mostly handled by the animation system later. I'm not sure if the EntitySound script is at all necessary at that point. For now, until the sound is done through the animation system, I think both of your solutions are valid.

Reply:
The `EntitySound` script itself will not become obsolete, but using it in code will, as its functions will be called through the animation system.
This is why I recommend just leaving the usage of `EntitySound` as it is on main right now, to avoid merge conflicts, as its current usage is removed later on.
Furthermore, changing the usage of it is not part of your task :)